scripts/src/java/org/oppia/android/scripts/coverage/CoverageRunner.kt

```diff
@@ -41,4 +53,176 @@ class CoverageRunner(
   ): List<String>? {
     return bazelClient.runCoverageForTestTarget(bazelTestTarget)
   }
+
+  private fun parseCoverageData(
+    coverageData: List<String>,
+    bazelTestTarget: String
+  ): CoverageReport {
+    var filePath = ""
+    var linesFound = 0
+    var linesHit = 0
+    val coveredLines = mutableListOf<CoveredLine>()
+    val branchCoverage = mutableListOf<BranchCoverage>()
+    val functionCoverage = mutableListOf<FunctionCoverage>()
+
+    var functionsFound = 0
+    var functionsHit = 0
+    var branchesFound = 0
+    var branchesHit = 0
+
+    var parseFile = false
+    val extractedFileName = "${extractTargetName(bazelTestTarget)}.kt"
+
+    coverageData.forEach { line ->
+      when {
+        // SF:<absolute path to the source file>
+        line.startsWith("SF:") -> {
+          val sourceFilePath = line.substringAfter("SF:")
+          if (sourceFilePath.substringAfterLast("/") == extractedFileName) {
+            filePath = line.substringAfter("SF:")
+            parseFile = true
+          } else {
+            parseFile = false
+          }
+        }
+        parseFile -> {
+          when {
+            // DA:<line number>,<execution count>
+            line.startsWith("DA:") -> {
+              val parts = line.substringAfter("DA:").split(",")
+              val lineNumber = parts[0].toInt()
+              val hitCount = parts[1].toInt()
+              val coverage =
+                if (hitCount > 0)
+                  Coverage.FULL
+                else
+                  Coverage.NONE
+              coveredLines.add(
+                CoveredLine.newBuilder()
+                  .setLineNumber(lineNumber)
+                  .setCoverage(coverage)
+                  .build()
+              )
```

Comment:
```suggestion
              val (lineNumStr, hitCountStr) = line.substringAfter("DA:").split(",")
              coveredLines += CoveredLine.newBuilder().apply {
                this.lineNumber= lineNumStr.toInt()
                this.coverage = if (hitCountStr.toInt() > 0) Coverage.FULL else Coverage.NONE
              }.build()
```

Destructuring and ``apply`` let you streamline this type of code a lot more, as an option.

Reply:
Thanks Ben! Updated the parsing logic.